src/sortedDict.jl

```diff
@@ -429,21 +122,21 @@ end
 
 function getkey{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}, k_, default_)
     i, exactfound = findkey(m.bt, convert(K,k_))
-    exactfound? m.bt.data[i].k : convert(K,default_)
+    exactfound? m.bt.data[i].k : convert(K, default_)
 end
 
 ## Function delete! deletes an item at a given 
 ## key
 
-function delete!{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}, k_)
-    i, exactfound = findkey(m.bt,convert(K,k_))
+@inline function delete!(m::SortedDict, k_)
+    i, exactfound = findkey(m.bt,convert(keytype(m),k_))
```

Comment:
Same here.  I'm guessing that you removed the type parameters under the assumption that `@inline` would work, but I don't believe that matters.  (Can someone more knowledgable verify this?)